### ISSUE TITLE
Tests failing in master (?)

### ISSUE THREAD

**Issue by bukzor** [JanusGraph/janusgraph#1086]
I ran `mvn clean install` per the instructions, but received two test errors. Am I holding it wrong?

https://paste.pound-python.org/show/QOrwKbY3G0F7blik5qQW/

I had this commit checked out at the time:
```
commit 7bef25948d520b8b064e3cce5c9b6ca7d4c41beb (origin/master, origin/HEAD, master)                                                                  
Author: Jan Jansen <jan.jansen@gdata.de>                                   
Date:   Wed May 16 08:12:19 2018 +0200                                     

    Fix order of TypeDefinitionCategory enum                               
    switch the config type of SCHEMA_CONSTRAINTS to GLOBAL_OFFLINE         
                                     
    Fixes #1058                      
                                     
    Signed-off-by: Jan Jansen <jan.jansen@gdata.de>                        

```

I'm going to try again on the 0.2 branch.

**Comment by mbrukman** [comment]
@bukzor, what's your environment (OS, version, Java version, etc.)?

As you can see, that same commit passed on the branch and on `master` on Travis CI:
https://travis-ci.org/JanusGraph/janusgraph/builds/379826873
but maybe your environment is different from what's running on Travis.

**Comment by bukzor** [comment]
Here's my environment, below, and here's the log of my attempt on 0.2 branch:
https://paste.pound-python.org/show/KMRcaXjwmsGQZf3kVfQq/

```
$ uname -a
Linux tortle-2016 4.13.0-36-lowlatency #40-Ubuntu SMP PREEMPT Fri Feb 16 21:12:41 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

$ git log -1
WARNING: terminal is not fully functional
commit 2a60728607d5da220986675533a1bf5dfa412ab5 (HEAD -> 0.2, origin/0.2)
Author: Chris Hupman <chupman@us.ibm.com>
Date:   Wed May 16 09:26:46 2018 -0700

    Fixed the rest of the javadoc breaking errors and enabled linting [skip ci]

    Signed-off-by: Chris Hupman <chupman@us.ibm.com>

$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 17.10
Release:        17.10
Codename:       artful

$ java --version
openjdk 9-Ubuntu
OpenJDK Runtime Environment (build 9-Ubuntu+0-9b181-4)
OpenJDK 64-Bit Server VM (build 9-Ubuntu+0-9b181-4, mixed mode)

$ dpkg -S $(readlink -f $(which java))
openjdk-9-jre-headless:amd64: /usr/lib/jvm/java-9-openjdk-amd64/bin/java

$ apt-cache show openjdk-9-jre-headless
Package: openjdk-9-jre-headless
Architecture: amd64
Version: 9~b181-4
Multi-Arch: same
Priority: optional
Section: universe/java
Source: openjdk-9
Origin: Ubuntu
Maintainer: OpenJDK Team <openjdk@lists.launchpad.net>
Bugs: https://bugs.launchpad.net/ubuntu/+filebug
Installed-Size: 264118
Provides: java-runtime-headless, java2-runtime-headless, java5-runtime-headless, java6-runtime-headless, java7-runtime-headless,
java8-runtime-headless, java9-runtime-headless
Depends: ca-certificates-java, java-common (>= 0.28), libcups2, liblcms2-2 (>= 2.2+git20110628), libjpeg8 (>= 8c), libnss3 (>=
2:3.17.1), util-linux (>= 2.26.2-4), libc6 (>= 2.14), libelf1 (>= 0.131), libfreetype6 (>= 2.2.1), libgcc1 (>= 1:3.4), libpcsclite1
(>= 1.3.0), libstdc++6 (>= 4.1.1), libx11-6, libxext6, libxi6, libxrender1, libxtst6, zlib1g (>= 1:1.1.4)
Suggests: libnss-mdns, fonts-dejavu-extra, fonts-ipafont-gothic, fonts-ipafont-mincho, ttf-wqy-microhei | ttf-wqy-zenhei,
fonts-indic
Breaks: oracle-java9-installer
Filename: pool/universe/o/openjdk-9/openjdk-9-jre-headless_9~b181-4_amd64.deb
Size: 116864384
MD5sum: 872c08e66cf8e4d7f7f8f9efce9759da
SHA1: 5d272a8c8ac089a4a8f712fd0e5495f745f66ac6
SHA256: 978f8fb22c499cb193d6ec473ef2b4a80d4213d7a785176131500aa184aa6396
Homepage: http://openjdk.java.net/
Description: OpenJDK Java runtime, using Hotspot JIT (headless)
Description-md5: 62c2c9689e617c48cc4480926d2ca785
```

**Comment by mbrukman** [comment]
Thanks for the additional details. We develop and test only with JDK 8 (see [`BUILDING.md`](https://github.com/JanusGraph/janusgraph/blob/master/BUILDING.md)), not JDK 9, so it's unfortunate, but not unexpected, that there are issues or incompatibilities with JDK 9.

There was a JDK-9-only issue reported (and fixed) in the past—see https://github.com/JanusGraph/janusgraph/issues/867 for more info—looks like it's not the only one.

Please try building with JDK 8 on your system; if that works, which I expect it will, it will be yet another JDK-9-only issue. If you can try to isolate the cause of it (and maybe even propose a fix), it would be much appreciated.

**Comment by bukzor** [comment]
Thanks. I believe I read those as minimum versions. Is the current intent
to support Java 9 as well? Your other ticket indicates that it essentially
works already.

**Comment by mbrukman** [comment]
@bukzor — sorry, Java 8 is meant to be "exactly" Java 8 rather than "minimum" as we do not test with Java 9, and as you can see, we've run into issues with it in the past, and haven't fixed all the issues.

The ticket I referenced was to fix one of the issues, and it may have worked for the person who filed it at the time; or, perhaps that was just the initial blocker, and the issue you ran into is the next blocker after fixing that one.

The plan is to support Java 9 in the future, but as you can see, it does not work yet, and since we don't have Java 9 as one of the Travis CI configs, it's possible to get regressions again.

Please consider either downgrading to Java 8 for the time being, or help us add Java 9 support and add a Travis CI build to ensure it doesn't regress in the future. Sorry for the trouble you ran into.

**Comment by bukzor** [comment]
Thanks. Will do. Just clarifying :)

On May 28, 2018 12:17 PM, "Misha Brukman" <notifications@github.com> wrote:

@bukzor <https://github.com/bukzor> — sorry, Java 8 is meant to be
"exactly" Java 8 rather than "minimum" as we do not test with Java 9, and
as you can see, we've run into issues with it in the past, and haven't
fixed all the issues.

The ticket I referenced was to fix one of the issues, and it may have
worked for the person who filed it at the time; or, perhaps that was just
the initial blocker, and the issue you ran into is the next blocker after
fixing that one.

The plan is to support Java 9 in the future, but as you can see, it does
not work yet, and since we don't have Java 9 as one of the Travis CI
configs, it's possible to get regressions again.

Please consider either downgrading to Java 8 for the time being, or help us
add Java 9 support and add a Travis CI build to ensure it doesn't regress
in the future. Sorry for the trouble you ran into.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/JanusGraph/janusgraph/issues/1086#issuecomment-392589679>,
or mute the thread

**Comment by chupman** [comment]
I just wanted to add a comment since I looked into what Java 9 support would entail awhile back. I didn't get very far once I saw the [Java SE Support roadmap](http://www.oracle.com/technetwork/java/javase/eol-135779.html). While end of support for Java 9 hasn't been announced anywhere I've seen, it was supposed to be out of support in March. 

I don't know what the JanusGraph direction will be in regards to Java 9 and 10, but I know personally I choose not to prioritize learning them since they only have a 6 month lifecycle. That being said I'm expecting migrating to 11 in a few months to be less than pleasant.
